ugs-core/src/com/willwinder/universalgcodesender/GrblController.java

```diff
@@ -477,6 +495,13 @@ public void toggleCheckMode() throws Exception {
             if (!"".equals(gcode)) {
                 GcodeCommand command = createCommand(gcode);
                 this.sendCommandImmediately(command);
+
+                if (getControlState() == COMM_CHECK) {
```

Comment:
This if/else should probably be in `handleStatusString` (i.e. in case a soft reset or external button press changes the state out of CHECK mode)

Reply:
This took some work because the GRBL issues a soft reset when exiting Check-mode clearing the previous state. So there is nothing to compare with when trying to reset the single step mode. I solved this by not clearing the old status if we are in Check-mode.